metadata/metadata.go

```diff
@@ -1848,78 +2014,88 @@ func isResourceReady(res Resource) bool {
 }
 
 type nameStream interface {
-	Recv() (*pb.Name, error)
+	Recv() (*pb.NameRequest, error)
 }
 
 type variantStream interface {
-	Recv() (*pb.NameVariant, error)
+	Recv() (*pb.NameVariantRequest, error)
 }
 
 type sendFn func(proto.Message) error
 
 type initParentFn func(name, variant string) Resource
 
 func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, init initParentFn) (*pb.Empty, error) {
-	serv.Logger.Info("Creating Generic Resource: ", res.ID().Name, res.ID().Variant)
+	logger := logging.GetLoggerFromContext(ctx).WithResource(res.ID().Type.ToLoggingResourceType(), res.ID().Name, res.ID().Variant)
+	logger.Debugw("Creating Generic Resource: ", res.ID().Name, res.ID().Variant)
 
 	id := res.ID()
-	if err := resourceNamedSafely(id); err != nil {
+	if err := resourceNamedSafely(id, logger); err != nil {
```

Comment:
should this take in the context instead of the logger?

Reply:
Actually I don't think it should take a context or a logger, its too small

Reply:
True, the error will just show in the function that calls it